File: sqlite3/ocaml/delete/sqlite3_delete.ml
(* ------------------------------------------------------------------- *)
(*
	sqlite3_delete.ml

					Jun/16/2011

*)
(* ------------------------------------------------------------------- *)
open Printf
open Sqlite3

exception Dummy
(* ------------------------------------------------------------------- *)
let sql_execute_proc = fun db sql_str -> 
	try
	match
	exec db sql_str ~cb:(fun _ _ -> print_endline "FOUND!"; raise Dummy)
	with
	| Rc.OK -> print_endline "OK"
	| _ -> assert false
	with xcp -> print_endline (Printexc.to_string xcp)
	
(* ------------------------------------------------------------------- *)
let sql_delete_proc = fun db id_in -> 
  let sql_str = sprintf
 "delete from cities where id = '" ^ id_in ^ "'"
 in
    print_endline sql_str;
    sql_execute_proc db sql_str

(* ------------------------------------------------------------------- *)
let _ = print_endline "*** 開始 ***"
let _ = printf "%s\n" Sys.argv.(1)
let _ = printf "%s\n" Sys.argv.(2)

let _ = let db = db_open Sys.argv.(1) in
		sql_delete_proc db (Sys.argv.(2))

let _ = print_endline "*** 終了 ***"
(* ------------------------------------------------------------------- *)

File: sqlite3/go/delete/sqlite3_delete.go
// ----------------------------------------------------------------
//
//	sqlite3_delete.go
//
//					May/19/2018
//
// ----------------------------------------------------------------
package main

import (
	"fmt"
	"os"
	"database/sql"
	_ "github.com/mattn/go-sqlite3"
)

// ----------------------------------------------------------------
func main() {
	fmt.Println ("*** 開始 ***")

	key_in := os.Args[1]

	fmt.Printf ("key_in = %s\n" , key_in)

	db_file := "/var/tmp/sqlite3/cities.db"
	db, err := sql.Open("sqlite3", db_file)
	if err != nil {
		fmt.Println ("*** error ***")
		fmt.Println(err)
		return
		}


	sql_str := "delete from cities WHERE id = ?"

	_,err = db.Exec (sql_str,key_in)
	if err != nil {
		fmt.Println(err)
	}

	fmt.Println ("*** 終了 ***")
}

// ----------------------------------------------------------------
